src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -1580,15 +1612,65 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 					break;
 				case IECoreVDB::VDBObjectTypeId :
 					h.append( m_volumePadding );
-
 					m_volume.hash( h );
 					break;
 				default :
+					if(
+						objectType == (IECore::TypeId)GafferScene::PreviewProceduralTypeId ||
+						IECore::RunTimeTyped::inheritsFrom( objectType, (IECore::TypeId)GafferScene::PreviewProceduralTypeId )
```

Comment:
Thinking about why we need two checks here ... the first check isn't an early exit, does `inheritsFrom( A, A )` not return true?  ( It looks like it should from a quick glance at Cortex? )

Reply:
Oh, wait, I was thinking about that backwards ... it is an early exit, so I guess this could be for hypothetical preformance reasons?  Don't mind me.

Reply:
You're right that this avoids a call to `inheritsFrom()` in some circumstances, but actually my motivation was that `inheritsFrom( A, A ) == false `.